### PR TITLE
[fabric] Ansible decoupling for chaincode ops

### PR DESCRIPTION
**Pull request by mgCepeda**
Signed-off-by: mgCepeda <marina.gomez.cepeda@accenture.com>

**Changelog**
- Add new yaml files in rol create/ca-tools
- Fix small bug in role delete/vault_secrets
- Update create/k8s_secrets, create/ca-tools, /create/chaincode/commit, create/chaincode/install, create/chaincode/invoke roles and /charts/commit_chaincode, /charts/install_chaincode/ charts

 

**Reviewed by**
@suvajit-sarkar 
@jagpreetsinghsasan 

 

**Linked issue**
#1720

### CODE REVIEW

File: platforms/hyperledger-fabric/charts/commit_chaincode/templates/commit_chaincode.yaml [hunk 1]
```diff
@@ -67,6 +71,15 @@ spec:
         args:
         - |-
           #!/usr/bin/env sh
+
+          formatCertificate () {
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/charts/commit_chaincode/templates/commit_chaincode.yaml:75]: Very good 👍 👍 👍

File: platforms/hyperledger-fabric/configuration/roles/create/ca-tools/peer/tasks/write.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#  Copyright Accenture. All Rights Reserved.
+#
+#  SPDX-License-Identifier: Apache-2.0
+##############################################################################################
+
+---
+############################################################################################
+# This task copy the peer certificate to the path provided in network.yaml
+- name: Copy msp cacerts from auto-generated path to given path
+  shell: |
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/ca-tools/peer/tasks/write.yaml:11]: This task should be done by pulling the certificate from vault and not using `kubectl cp` cmd

File: platforms/hyperledger-fabric/configuration/roles/create/ca-tools/peer/tasks/write.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  when: 
+    - item.org_status == 'new'
+    - org_peer.peerstatus is not defined or org_peer.peerstatus == 'new'
+    - network.env.proxy != 'none'
+    - org_peer.name == peer.name
+
+############################################################################################
+# This task copy the peer certificate to the path provided in network.yaml (when proxy is none)
+- name: Copy msp cacerts from auto-generated path to given path
+  shell: |
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/create/ca-tools/peer/tasks/write.yaml:26]: This task should be done by pulling the certificate from vault and not using `kubectl cp` cmd

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml [hunk 4]
```diff
@@ -117,16 +117,34 @@ network:
...
-      - carrier
-      - warehouse
-      - manufacturer
-      - store
-      corepeerAddress:
-      - peer0.carrier-net.hf.demo.aws.blockchaincloudpoc.com:8443
-      - peer0.warehouse-net.hf.demo.aws.blockchaincloudpoc.com:8443
-      - peer0.manufacturer-net.hf.demo.aws.blockchaincloudpoc.com:8443
-      - peer0.store-net.hf.demo.aws.blockchaincloudpoc.com:8443
+    - organization:
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml:120]: Write a comment here saying `Only one peer per org required for endorsement`

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml [hunk 5]
```diff
@@ -157,12 +175,20 @@ network:
...
       ordererAddress: orderer1.org1ambassador.blockchaincloudpoc.com:8443
     endorsers:
-      name:
-      - carrier
-      - store
-      corepeerAddress:
-      - peer0.carrier-net.hf.demo.aws.blockchaincloudpoc.com:8443
-      - peer0.store-net.hf.demo.aws.blockchaincloudpoc.com:8443
+    - organization:
+      name: carrier
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml:179]: Write a comment here saying `Only one peer per org required for endorsement`

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml [hunk 6]
```diff
@@ -114,14 +114,27 @@ network:
...
-      name:
-      - carrier
-      - warehouse
-      - manufacturer
-      corepeerAddress:
-      - peer0.carrier-net.org3ambassador.blockchaincloudpoc.com:8443
-      - peer0.warehouse-net.org4ambassador.blockchaincloudpoc.com:8443
-      - peer0.manufacturer-net.org2ambassador.blockchaincloudpoc.com:8443
+    - organization:
+      name: carrier
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml:118]: Write a comment here saying `Only one peer per org required for endorsement`

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml [hunk 7]
```diff
@@ -109,14 +109,27 @@ network:
...
     endorsers:
-      name:
-      - carrier
-      - warehouse
-      - manufacturer
-      corepeerAddress:
-      - peer0.carrier-net.org3ambassador.blockchaincloudpoc.com:8443
-      - peer0.warehouse-net.org4ambassador.blockchaincloudpoc.com:8443
-      - peer0.manufacturer-net.org2ambassador.blockchaincloudpoc.com:8443
+    - organization:
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml:112]: Write a comment here saying `Only one peer per org required for endorsement`

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft.yaml [hunk 8]
```diff
@@ -116,16 +116,34 @@ network:
...
-      - carrier
-      - warehouse
-      - manufacturer
-      - store
-      corepeerAddress:
-      - peer0.carrier-net.hf.demo.aws.blockchaincloudpoc.com:8443
-      - peer0.warehouse-net.hf.demo.aws.blockchaincloudpoc.com:8443
-      - peer0.manufacturer-net.hf.demo.aws.blockchaincloudpoc.com:8443
-      - peer0.store-net.hf.demo.aws.blockchaincloudpoc.com:8443
+    - organization:
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft.yaml:119]: Write a comment here saying `Only one peer per org required for endorsement`

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]

**Review by jagpreetsinghsasan — APPROVED** [review]

**Review by sownak — APPROVED** [review]

**Pull request merged**